ext/mysqli/tests/mysqli_change_user_new.phpt

```diff
@@ -11,6 +11,9 @@ if (!$link = my_mysqli_connect($host, $user, $passwd, $db, $port, $socket))
 
 if (mysqli_get_server_version($link) < 50600)
     die("SKIP For MySQL >= 5.6.0");
+
+if (mysqli_get_server_version($link) >= 10_00_00)
+    die("SKIP Not applicable for MariaDB");
```

Comment:
Note that this test has some variations. Would the mysqli_change_user_old.phpt version be applicable to MariaDB, maybe?

Reply:
Ok, I admit I jumped to conclusions on this one. Either the test case is wrong or the implementation is broken. If `COM_CHANGE_USER` fails then the connection is not reset on MariaDB. I don't think it should be reset, but maybe I am wrong. 
For the moment I will revert that commit, maybe someone will come up with a better way how to fix this test case.

Reply:
I believe mysqli_change_user_old.phpt is testing exactly the case where changing the user does not reset the connection.

Reply:
And I assumed that mysqli_change_user_old.phpt applies to old MySQL and MariaDB. mysqli_change_user_new.phpt should only be applicable to MySQL 5.6+

Reply:
To be clear, my suggestion is to change this test as you already did, and adjust the old.phpt test to not skip on >= 100000. Assuming it does pass in that case. If it doesn't pass, then this is probably not important enough to bother with.

Reply:
Ok, now I got it. I completely missed that. Sorry. I have added a new commit now which should fix the old test case too.